driver/pelican/types/thermSchedule.go

```diff
@@ -0,0 +1,246 @@
+	GroupId     string         `json:"groupId"`
+	Permissions string         `json:"permissions"`
+}
+
+type thermIDChild struct {
+	Id          string `json:"id"`
+	Permissions string `json:"permissions"`
+}
+
+// Thermostat Settings Structs
+type settingsRequest struct {
+	Epnum    float64         `json:"epnum"`
+	Id       string          `json:"id"`
+	Nodename string          `json:"nodename"`
+	Userdata settingsWrapper `json:"userdata"`
+}
+
+type settingsWrapper struct {
+	Epnum    int64  `json:"epnum"`
+	Fan      string `json:"fan"`
+	Nodename string `json:"nodename"`
+	Repeat   string `json:"repeat"`
+}
+
+// Thermostat Schedule By Day Decoding Structs
+type scheduleRequest struct {
+	ClientData scheduleSetTimes `json:"clientdata"`
+}
+
+type scheduleSetTimes struct {
+	SetTimes []scheduleTimeBlock `json:"setTimes"`
+}
+
+type scheduleTimeBlock struct {
+	StartValue    string `json:"startValue"`
+	SystemDisplay string `json:"systemDisplay"`
+	HeatSetting   int64  `json:"heatSetting"`
+	CoolSetting   int64  `json:"coolSetting"`
+}
+
+// Thermostat Schedule By Day Result Structs
+type ThermostatSchedule struct {
+	DaySchedules map[string]ThermostatDaySchedule
+}
+
+type ThermostatDaySchedule struct {
+	Blocks []ThermostatBlockSchedule
+}
+
+type ThermostatBlockSchedule struct {
+	CoolSetting int64
+	HeatSetting int64
+	System      string
+	Time        string
+}
+
+func (pel *Pelican) GetSchedule(sitename string) (map[string]ThermostatSchedule, error) {
+	// Retrieve Login Authentication Cookies
+	loginInfo := map[string]interface{}{
+		"username": pel.username,
+		"password": pel.password,
+		"sitename": sitename,
+	}
+	respLogin, _, errsLogin := gorequest.New().Post(fmt.Sprintf("https://%s.officeclimatecontrol.net/#_loginPage", sitename)).Type("form").Send(loginInfo).End()
+	if (errsLogin != nil) || (respLogin.StatusCode != 200) {
+		return nil, fmt.Errorf("Error logging into climate control website: %v", errsLogin)
+	}
+	cookies := (*http.Response)(respLogin).Cookies()
+	cookie := cookies[0]
+
+	// Retrieve Thermostat IDs within given sitename
+	respTherms, _, errsTherms := gorequest.New().Get(fmt.Sprintf("https://%s.officeclimatecontrol.net/ajaxSchedule.cgi?request=getResourcesExtended&resourceType=Thermostats", sitename)).Type("form").AddCookie(cookie).End()
+	if (errsTherms != nil) || (respTherms.StatusCode != 200) {
+		return nil, fmt.Errorf("Error retrieving Thermostat IDs: %v", errsTherms)
+	}
+	var result thermIDRequest
+	decoder := json.NewDecoder(respTherms.Body)
+	if decodeError := decoder.Decode(&result); decodeError != nil {
+		return nil, fmt.Errorf("Failed to decode Thermostat ID response JSON: %v\n", decodeError)
+	}
+	thermostatIDs := result.Resources[0].Children
+
+	// Construct Weekly Schedules for each Thermostat ID
+	schedules := make(map[string]ThermostatSchedule, len(thermostatIDs))
+	for _, child := range thermostatIDs {
+		week := []string{"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"}
```

Comment:
We can make this a `const` declared at the top of the file.

Reply:
Oops, yeah make sense. Well if you want you can declare it as a`var` at the top. I think the `pelican.go` has some similar stuff for handling the different thermostat modes.